cartoframes/context.py

```diff
@@ -216,6 +217,13 @@ def write(self, df, table_name, temp_dir=CACHE_DIR,
         if table_name != norm_colname(table_name):
             table_name = norm_colname(table_name)
             warn('Table will be named `{}`'.format(table_name))
+
+        # Issue warning if the index is anything but the Pandas default range index
+        if (df.index != np.arange(len(df))).any():
```

Comment:
Instead of creating a potentially huge array with with `np.arange`, we could rely on the pandas `RangeIndex` object for this piece:
```python
if df.index.equals(pd.core.indexes.range.RangeIndex(0, len(df), 1)):
    <do stuff>
```